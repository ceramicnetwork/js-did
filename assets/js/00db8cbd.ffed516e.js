"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[6258],{4852:(e,t,r)=>{r.d(t,{Zo:()=>s,kt:()=>k});var n=r(9231);function a(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}function i(e,t){var r=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),r.push.apply(r,n)}return r}function o(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{};t%2?i(Object(r),!0).forEach((function(t){a(e,t,r[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(r)):i(Object(r)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(r,t))}))}return e}function c(e,t){if(null==e)return{};var r,n,a=function(e,t){if(null==e)return{};var r,n,a={},i=Object.keys(e);for(n=0;n<i.length;n++)r=i[n],t.indexOf(r)>=0||(a[r]=e[r]);return a}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)r=i[n],t.indexOf(r)>=0||Object.prototype.propertyIsEnumerable.call(e,r)&&(a[r]=e[r])}return a}var p=n.createContext({}),l=function(e){var t=n.useContext(p),r=t;return e&&(r="function"==typeof e?e(t):o(o({},t),e)),r},s=function(e){var t=l(e.components);return n.createElement(p.Provider,{value:t},e.children)},u="mdxType",f={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},d=n.forwardRef((function(e,t){var r=e.components,a=e.mdxType,i=e.originalType,p=e.parentName,s=c(e,["components","mdxType","originalType","parentName"]),u=l(r),d=a,k=u["".concat(p,".").concat(d)]||u[d]||f[d]||i;return r?n.createElement(k,o(o({ref:t},s),{},{components:r})):n.createElement(k,o({ref:t},s))}));function k(e,t){var r=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var i=r.length,o=new Array(i);o[0]=d;var c={};for(var p in t)hasOwnProperty.call(t,p)&&(c[p]=t[p]);c.originalType=e,c[u]="string"==typeof e?e:a,o[1]=c;for(var l=2;l<i;l++)o[l]=r[l];return n.createElement.apply(null,o)}return n.createElement.apply(null,r)}d.displayName="MDXCreateElement"},1956:(e,t,r)=>{r.r(t),r.d(t,{assets:()=>p,contentTitle:()=>o,default:()=>f,frontMatter:()=>i,metadata:()=>c,toc:()=>l});var n=r(5799),a=(r(9231),r(4852));const i={id:"pkh_stacks.SignatureData",title:"Interface: SignatureData",custom_edit_url:null},o=void 0,c={unversionedId:"api/interfaces/pkh_stacks.SignatureData",id:"api/interfaces/pkh_stacks.SignatureData",title:"Interface: SignatureData",description:"pkh-stacks.SignatureData",source:"@site/docs/api/interfaces/pkh_stacks.SignatureData.md",sourceDirName:"api/interfaces",slug:"/api/interfaces/pkh_stacks.SignatureData",permalink:"/docs/api/interfaces/pkh_stacks.SignatureData",draft:!1,editUrl:null,tags:[],version:"current",frontMatter:{id:"pkh_stacks.SignatureData",title:"Interface: SignatureData",custom_edit_url:null}},p={},l=[{value:"Properties",id:"properties",level:2},{value:"publicKey",id:"publickey",level:3},{value:"signature",id:"signature",level:3}],s={toc:l},u="wrapper";function f(e){let{components:t,...r}=e;return(0,a.kt)(u,(0,n.Z)({},s,r,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("p",null,(0,a.kt)("a",{parentName:"p",href:"/docs/api/modules/pkh_stacks"},"pkh-stacks"),".SignatureData"),(0,a.kt)("h2",{id:"properties"},"Properties"),(0,a.kt)("h3",{id:"publickey"},"publicKey"),(0,a.kt)("p",null,"\u2022 ",(0,a.kt)("strong",{parentName:"p"},"publicKey"),": ",(0,a.kt)("inlineCode",{parentName:"p"},"string")),(0,a.kt)("hr",null),(0,a.kt)("h3",{id:"signature"},"signature"),(0,a.kt)("p",null,"\u2022 ",(0,a.kt)("strong",{parentName:"p"},"signature"),": ",(0,a.kt)("inlineCode",{parentName:"p"},"string")))}f.isMDXComponent=!0}}]);