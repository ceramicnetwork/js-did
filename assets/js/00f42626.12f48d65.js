"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[2118],{4852:(e,t,r)=>{r.d(t,{Zo:()=>m,kt:()=>s});var n=r(9231);function a(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}function o(e,t){var r=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),r.push.apply(r,n)}return r}function i(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{};t%2?o(Object(r),!0).forEach((function(t){a(e,t,r[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(r)):o(Object(r)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(r,t))}))}return e}function p(e,t){if(null==e)return{};var r,n,a=function(e,t){if(null==e)return{};var r,n,a={},o=Object.keys(e);for(n=0;n<o.length;n++)r=o[n],t.indexOf(r)>=0||(a[r]=e[r]);return a}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(n=0;n<o.length;n++)r=o[n],t.indexOf(r)>=0||Object.prototype.propertyIsEnumerable.call(e,r)&&(a[r]=e[r])}return a}var u=n.createContext({}),l=function(e){var t=n.useContext(u),r=t;return e&&(r="function"==typeof e?e(t):i(i({},t),e)),r},m=function(e){var t=l(e.components);return n.createElement(u.Provider,{value:t},e.children)},d="mdxType",c={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},h=n.forwardRef((function(e,t){var r=e.components,a=e.mdxType,o=e.originalType,u=e.parentName,m=p(e,["components","mdxType","originalType","parentName"]),d=l(r),h=a,s=d["".concat(u,".").concat(h)]||d[h]||c[h]||o;return r?n.createElement(s,i(i({ref:t},m),{},{components:r})):n.createElement(s,i({ref:t},m))}));function s(e,t){var r=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var o=r.length,i=new Array(o);i[0]=h;var p={};for(var u in t)hasOwnProperty.call(t,u)&&(p[u]=t[u]);p.originalType=e,p[d]="string"==typeof e?e:a,i[1]=p;for(var l=2;l<o;l++)i[l]=r[l];return n.createElement.apply(null,i)}return n.createElement.apply(null,r)}h.displayName="MDXCreateElement"},7229:(e,t,r)=>{r.r(t),r.d(t,{assets:()=>u,contentTitle:()=>i,default:()=>c,frontMatter:()=>o,metadata:()=>p,toc:()=>l});var n=r(5799),a=(r(9231),r(4852));const o={id:"pkh_ethereum.EthereumNodeAuth",title:"Namespace: EthereumNodeAuth",custom_edit_url:null},i=void 0,p={unversionedId:"api/namespaces/pkh_ethereum.EthereumNodeAuth",id:"api/namespaces/pkh_ethereum.EthereumNodeAuth",title:"Namespace: EthereumNodeAuth",description:"pkh-ethereum.EthereumNodeAuth",source:"@site/docs/api/namespaces/pkh_ethereum.EthereumNodeAuth.md",sourceDirName:"api/namespaces",slug:"/api/namespaces/pkh_ethereum.EthereumNodeAuth",permalink:"/docs/api/namespaces/pkh_ethereum.EthereumNodeAuth",draft:!1,editUrl:null,tags:[],version:"current",frontMatter:{id:"pkh_ethereum.EthereumNodeAuth",title:"Namespace: EthereumNodeAuth",custom_edit_url:null}},u={},l=[{value:"Functions",id:"functions",level:2},{value:"getAuthMethod",id:"getauthmethod",level:3},{value:"Parameters",id:"parameters",level:4},{value:"Returns",id:"returns",level:4}],m={toc:l},d="wrapper";function c(e){let{components:t,...r}=e;return(0,a.kt)(d,(0,n.Z)({},m,r,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("p",null,(0,a.kt)("a",{parentName:"p",href:"/docs/api/modules/pkh_ethereum"},"pkh-ethereum"),".EthereumNodeAuth"),(0,a.kt)("h2",{id:"functions"},"Functions"),(0,a.kt)("h3",{id:"getauthmethod"},"getAuthMethod"),(0,a.kt)("p",null,"\u25b8 ",(0,a.kt)("strong",{parentName:"p"},"getAuthMethod"),"(",(0,a.kt)("inlineCode",{parentName:"p"},"ethProvider"),", ",(0,a.kt)("inlineCode",{parentName:"p"},"account"),", ",(0,a.kt)("inlineCode",{parentName:"p"},"appName"),"): ",(0,a.kt)("inlineCode",{parentName:"p"},"Promise"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"AuthMethod"),">"),(0,a.kt)("p",null,"Get a configured authMethod for an Ethereum account in a Node based environment"),(0,a.kt)("h4",{id:"parameters"},"Parameters"),(0,a.kt)("table",null,(0,a.kt)("thead",{parentName:"table"},(0,a.kt)("tr",{parentName:"thead"},(0,a.kt)("th",{parentName:"tr",align:"left"},"Name"),(0,a.kt)("th",{parentName:"tr",align:"left"},"Type"))),(0,a.kt)("tbody",{parentName:"table"},(0,a.kt)("tr",{parentName:"tbody"},(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"ethProvider")),(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"any"))),(0,a.kt)("tr",{parentName:"tbody"},(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"account")),(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"AccountId"))),(0,a.kt)("tr",{parentName:"tbody"},(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"appName")),(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"string"))))),(0,a.kt)("h4",{id:"returns"},"Returns"),(0,a.kt)("p",null,(0,a.kt)("inlineCode",{parentName:"p"},"Promise"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"AuthMethod"),">"))}c.isMDXComponent=!0}}]);