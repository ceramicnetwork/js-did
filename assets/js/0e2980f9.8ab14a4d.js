"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[6650],{4852:(e,t,r)=>{r.d(t,{Zo:()=>s,kt:()=>u});var n=r(9231);function a(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}function i(e,t){var r=Object.keys(e);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);t&&(n=n.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),r.push.apply(r,n)}return r}function l(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{};t%2?i(Object(r),!0).forEach((function(t){a(e,t,r[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(r)):i(Object(r)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(r,t))}))}return e}function o(e,t){if(null==e)return{};var r,n,a=function(e,t){if(null==e)return{};var r,n,a={},i=Object.keys(e);for(n=0;n<i.length;n++)r=i[n],t.indexOf(r)>=0||(a[r]=e[r]);return a}(e,t);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(n=0;n<i.length;n++)r=i[n],t.indexOf(r)>=0||Object.prototype.propertyIsEnumerable.call(e,r)&&(a[r]=e[r])}return a}var p=n.createContext({}),d=function(e){var t=n.useContext(p),r=t;return e&&(r="function"==typeof e?e(t):l(l({},t),e)),r},s=function(e){var t=d(e.components);return n.createElement(p.Provider,{value:t},e.children)},k="mdxType",m={inlineCode:"code",wrapper:function(e){var t=e.children;return n.createElement(n.Fragment,{},t)}},c=n.forwardRef((function(e,t){var r=e.components,a=e.mdxType,i=e.originalType,p=e.parentName,s=o(e,["components","mdxType","originalType","parentName"]),k=d(r),c=a,u=k["".concat(p,".").concat(c)]||k[c]||m[c]||i;return r?n.createElement(u,l(l({ref:t},s),{},{components:r})):n.createElement(u,l({ref:t},s))}));function u(e,t){var r=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var i=r.length,l=new Array(i);l[0]=c;var o={};for(var p in t)hasOwnProperty.call(t,p)&&(o[p]=t[p]);o.originalType=e,o[k]="string"==typeof e?e:a,l[1]=o;for(var d=2;d<i;d++)l[d]=r[d];return n.createElement.apply(null,l)}return n.createElement.apply(null,r)}c.displayName="MDXCreateElement"},6756:(e,t,r)=>{r.r(t),r.d(t,{assets:()=>p,contentTitle:()=>l,default:()=>m,frontMatter:()=>i,metadata:()=>o,toc:()=>d});var n=r(5799),a=(r(9231),r(4852));const i={id:"didtools_key_secp256k1.Secp256k1Provider",title:"Class: Secp256k1Provider",custom_edit_url:null},l=void 0,o={unversionedId:"api/classes/didtools_key_secp256k1.Secp256k1Provider",id:"api/classes/didtools_key_secp256k1.Secp256k1Provider",title:"Class: Secp256k1Provider",description:"@didtools/key-secp256k1.Secp256k1Provider",source:"@site/docs/api/classes/didtools_key_secp256k1.Secp256k1Provider.md",sourceDirName:"api/classes",slug:"/api/classes/didtools_key_secp256k1.Secp256k1Provider",permalink:"/docs/api/classes/didtools_key_secp256k1.Secp256k1Provider",draft:!1,editUrl:null,tags:[],version:"current",frontMatter:{id:"didtools_key_secp256k1.Secp256k1Provider",title:"Class: Secp256k1Provider",custom_edit_url:null}},p={},d=[{value:"Implements",id:"implements",level:2},{value:"Constructors",id:"constructors",level:2},{value:"constructor",id:"constructor",level:3},{value:"Parameters",id:"parameters",level:4},{value:"Returns",id:"returns",level:4},{value:"Properties",id:"properties",level:2},{value:"_handle",id:"_handle",level:3},{value:"Accessors",id:"accessors",level:2},{value:"isDidProvider",id:"isdidprovider",level:3},{value:"Returns",id:"returns-1",level:4},{value:"Methods",id:"methods",level:2},{value:"send",id:"send",level:3},{value:"Type parameters",id:"type-parameters",level:4},{value:"Parameters",id:"parameters-1",level:4},{value:"Returns",id:"returns-2",level:4},{value:"Implementation of",id:"implementation-of",level:4}],s={toc:d},k="wrapper";function m(e){let{components:t,...r}=e;return(0,a.kt)(k,(0,n.Z)({},s,r,{components:t,mdxType:"MDXLayout"}),(0,a.kt)("p",null,(0,a.kt)("a",{parentName:"p",href:"/docs/api/modules/didtools_key_secp256k1"},"@didtools/key-secp256k1"),".Secp256k1Provider"),(0,a.kt)("h2",{id:"implements"},"Implements"),(0,a.kt)("ul",null,(0,a.kt)("li",{parentName:"ul"},(0,a.kt)("inlineCode",{parentName:"li"},"DIDProvider"))),(0,a.kt)("h2",{id:"constructors"},"Constructors"),(0,a.kt)("h3",{id:"constructor"},"constructor"),(0,a.kt)("p",null,"\u2022 ",(0,a.kt)("strong",{parentName:"p"},"new Secp256k1Provider"),"(",(0,a.kt)("inlineCode",{parentName:"p"},"seed"),"): ",(0,a.kt)("a",{parentName:"p",href:"/docs/api/classes/didtools_key_secp256k1.Secp256k1Provider"},(0,a.kt)("inlineCode",{parentName:"a"},"Secp256k1Provider"))),(0,a.kt)("h4",{id:"parameters"},"Parameters"),(0,a.kt)("table",null,(0,a.kt)("thead",{parentName:"table"},(0,a.kt)("tr",{parentName:"thead"},(0,a.kt)("th",{parentName:"tr",align:"left"},"Name"),(0,a.kt)("th",{parentName:"tr",align:"left"},"Type"))),(0,a.kt)("tbody",{parentName:"table"},(0,a.kt)("tr",{parentName:"tbody"},(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"seed")),(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"Uint8Array"))))),(0,a.kt)("h4",{id:"returns"},"Returns"),(0,a.kt)("p",null,(0,a.kt)("a",{parentName:"p",href:"/docs/api/classes/didtools_key_secp256k1.Secp256k1Provider"},(0,a.kt)("inlineCode",{parentName:"a"},"Secp256k1Provider"))),(0,a.kt)("h2",{id:"properties"},"Properties"),(0,a.kt)("h3",{id:"_handle"},"_","handle"),(0,a.kt)("p",null,"\u2022 ",(0,a.kt)("strong",{parentName:"p"},"_","handle"),": ",(0,a.kt)("inlineCode",{parentName:"p"},"SendRequestFunc"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"DIDProviderMethods"),">"),(0,a.kt)("h2",{id:"accessors"},"Accessors"),(0,a.kt)("h3",{id:"isdidprovider"},"isDidProvider"),(0,a.kt)("p",null,"\u2022 ",(0,a.kt)("inlineCode",{parentName:"p"},"get")," ",(0,a.kt)("strong",{parentName:"p"},"isDidProvider"),"(): ",(0,a.kt)("inlineCode",{parentName:"p"},"boolean")),(0,a.kt)("h4",{id:"returns-1"},"Returns"),(0,a.kt)("p",null,(0,a.kt)("inlineCode",{parentName:"p"},"boolean")),(0,a.kt)("h2",{id:"methods"},"Methods"),(0,a.kt)("h3",{id:"send"},"send"),(0,a.kt)("p",null,"\u25b8 ",(0,a.kt)("strong",{parentName:"p"},"send"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"Name"),">","(",(0,a.kt)("inlineCode",{parentName:"p"},"msg"),"): ",(0,a.kt)("inlineCode",{parentName:"p"},"Promise"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"null")," ","|"," ",(0,a.kt)("inlineCode",{parentName:"p"},"RPCResponse"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"DIDProviderMethods"),", ",(0,a.kt)("inlineCode",{parentName:"p"},"Name"),">",">"),(0,a.kt)("h4",{id:"type-parameters"},"Type parameters"),(0,a.kt)("table",null,(0,a.kt)("thead",{parentName:"table"},(0,a.kt)("tr",{parentName:"thead"},(0,a.kt)("th",{parentName:"tr",align:"left"},"Name"),(0,a.kt)("th",{parentName:"tr",align:"left"},"Type"))),(0,a.kt)("tbody",{parentName:"table"},(0,a.kt)("tr",{parentName:"tbody"},(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"Name")),(0,a.kt)("td",{parentName:"tr",align:"left"},"extends keyof ",(0,a.kt)("inlineCode",{parentName:"td"},"DIDProviderMethods"))))),(0,a.kt)("h4",{id:"parameters-1"},"Parameters"),(0,a.kt)("table",null,(0,a.kt)("thead",{parentName:"table"},(0,a.kt)("tr",{parentName:"thead"},(0,a.kt)("th",{parentName:"tr",align:"left"},"Name"),(0,a.kt)("th",{parentName:"tr",align:"left"},"Type"))),(0,a.kt)("tbody",{parentName:"table"},(0,a.kt)("tr",{parentName:"tbody"},(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"msg")),(0,a.kt)("td",{parentName:"tr",align:"left"},(0,a.kt)("inlineCode",{parentName:"td"},"RPCRequest"),"<",(0,a.kt)("inlineCode",{parentName:"td"},"DIDProviderMethods"),", ",(0,a.kt)("inlineCode",{parentName:"td"},"Name"),">")))),(0,a.kt)("h4",{id:"returns-2"},"Returns"),(0,a.kt)("p",null,(0,a.kt)("inlineCode",{parentName:"p"},"Promise"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"null")," ","|"," ",(0,a.kt)("inlineCode",{parentName:"p"},"RPCResponse"),"<",(0,a.kt)("inlineCode",{parentName:"p"},"DIDProviderMethods"),", ",(0,a.kt)("inlineCode",{parentName:"p"},"Name"),">",">"),(0,a.kt)("h4",{id:"implementation-of"},"Implementation of"),(0,a.kt)("p",null,"DIDProvider.send"))}m.isMDXComponent=!0}}]);