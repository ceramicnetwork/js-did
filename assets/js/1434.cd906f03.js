"use strict";(self.webpackChunkwebsite=self.webpackChunkwebsite||[]).push([[1434],{1434:(e,s,b)=>{b.r(s)}}]);