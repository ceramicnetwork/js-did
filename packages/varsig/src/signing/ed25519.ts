import { MAGIC } from '../magic.js'
import { SigningAlgo } from '../signing.js'
import { ed25519 } from '@noble/curves/ed25519'
import type { BytesTape } from '../bytes-tape.js'

const SIGIL = MAGIC.ED25519

// @ts-ignore
function prepareVerifier(tape: BytesTape): SigningAlgo {
  return {
    kind: SIGIL,
    // eslint-disable-next-line @typescript-eslint/require-await
    verify: async (input, signature, verificationKey): Promise<boolean> => {
      return ed25519.verify(signature, input, verificationKey)
    },
  }
}

function readSignature(tape: BytesTape): Uint8Array {
  return tape.read(64)
}

export const ED25519 = { SIGIL, prepareVerifier, readSignature }
