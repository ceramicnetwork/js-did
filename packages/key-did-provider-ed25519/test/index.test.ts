import * as u8a from 'uint8arrays'
import { verifyJWS, createJWE, x25519Encrypter } from 'did-jwt'
import { randomBytes } from '@stablelib/random'
import { KeyPair, generateKeyPairFromSeed, convertPublicKeyToX25519 } from '@stablelib/ed25519'
import type { GeneralJWS } from 'dids'

import { encodeDID, Ed25519Provider } from '../src/index.js'

const b64urlToObj = (s: string): Record<string, any> =>
  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
  JSON.parse(u8a.toString(u8a.fromString(s, 'base64url')))

describe('key-did-provider-ed25519', () => {
  let provider: Ed25519Provider
  let did: string
  let kp: KeyPair

  beforeAll(() => {
    const seed = randomBytes(32)
    kp = generateKeyPairFromSeed(seed)
    provider = new Ed25519Provider(seed)
    did = encodeDID(kp.publicKey)
  })

  it('encodeDID', () => {
    const pubkey = u8a.fromString(
      'd713cb7f8624d8648496e01010f2bd72f0dcbbdecdb7036f38c20475f5f429bf',
      'base16'
    )
    expect(encodeDID(pubkey)).toMatchSnapshot()
  })

  it('has isDidProvider property', () => {
    expect(provider.isDidProvider).toEqual(true)
  })

  it('authenticates correctly', async () => {
    const nonce = 'adfberg'
    const aud = 'https://my.app'
    const paths = ['a', 'b']
    const resp = await provider.send({
      jsonrpc: '2.0',
      id: 0,
      method: 'did_authenticate',
      params: { nonce, aud, paths },
    })
    const jws = resp?.result as GeneralJWS
    const payload = b64urlToObj(jws.payload)
    const header = b64urlToObj(jws.signatures[0].protected)
    expect(payload.aud).toEqual(aud)
    expect(payload.nonce).toEqual(nonce)
    expect(payload.paths).toEqual(paths)
    expect(payload.did).toEqual(did)
    expect(payload.exp).toBeGreaterThan(Date.now() / 1000)
    expect(header.kid).toEqual(expect.stringContaining(did))
    expect(header.alg).toEqual('EdDSA')
  })

  it('signs JWS properly', async () => {
    const payload = { foo: 'bar' }
    const prot = { bar: 'baz' }
    const res = await provider.send({
      jsonrpc: '2.0',
      id: 0,
      method: 'did_createJWS',
      params: { payload, protected: prot, did },
    })
    const pubkey = {
      id: '',
      type: '',
      controller: '',
      publicKeyBase64: u8a.toString(kp.publicKey, 'base64pad'),
    }
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unsafe-assignment
    const gjws = res?.result?.jws as GeneralJWS
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    const jws = [gjws.signatures[0].protected, gjws.payload, gjws.signatures[0].signature].join('.')
    expect(verifyJWS(jws, pubkey)).toEqual(pubkey)
  })

  it('decrypts JWE properly', async () => {
    const encrypter = x25519Encrypter(convertPublicKeyToX25519(kp.publicKey))
    const cleartext = randomBytes(123)
    const jwe = await createJWE(cleartext, [encrypter])
    const res = await provider.send({
      jsonrpc: '2.0',
      id: 0,
      method: 'did_decryptJWE',
      params: { jwe },
    })
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    expect(res?.result?.cleartext as string).toEqual(u8a.toString(cleartext, 'base64pad'))
  })

  it('thows if fails to decrypt JWE', async () => {
    const encrypter = x25519Encrypter(randomBytes(32))
    const cleartext = randomBytes(123)
    const jwe = await createJWE(cleartext, [encrypter])
    const res = await provider.send({
      jsonrpc: '2.0',
      id: 0,
      method: 'did_decryptJWE',
      params: { jwe },
    })
    expect(res).toEqual({
      jsonrpc: '2.0',
      id: 0,
      error: {
        data: undefined,
        code: -32000,
        message: 'failure: Failed to decrypt',
      },
    })
  })
})
