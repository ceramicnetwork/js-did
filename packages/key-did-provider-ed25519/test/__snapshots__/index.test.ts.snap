// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`key-did-provider-ed25519 encodeDID 1`] = `"did:key:z6MktvqCyLxTsXUH1tUZncNdVeEZ7hNh7npPRbUU27GTrYb8"`;
